### PR TITLE
HTML5-Form

### PR DESCRIPTION
**Pull request by Zihfred**
### HTML5-Form
[Demo](https://zihfred.github.io/html-5-form/scr/) |
[Code base](https://github.com/Zihfred/html-5-form/tree/master/scr)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/Zihfred/html5-form/index.html [hunk 1]
```diff
@@ -0,0 +1,591 @@
...
+    <link rel="stylesheet" href="style/main.css">
+    <link href="https://fonts.googleapis.com/css?family=Lato" rel="stylesheet">
+    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.1/css/all.css"
+          integrity="sha384-fnmOCqbTlWIlj8LyTjo7mOUStjsKC4pOpQbqyi7RrhN7udi9RwhKkMHpvLbHG9Sr"
+          crossorigin="anonymous">
+    <link rel="stylesheet" href="style/responsive.css">
+</head>
+<body>
+<div class="wrapper">
+    <form action="URL">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: Add the border according to the mockup.

File: submissions/Zihfred/html5-form/index.html [hunk 2]
```diff
@@ -0,0 +1,591 @@
...
+        <div class="formHeader">
+            <div class="yourOrders">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: For this case better use `table`.

File: submissions/Zihfred/html5-form/index.html [hunk 3]
```diff
@@ -0,0 +1,591 @@
...
+                    <h3 class="youOrdersItem__text">
+                        delivery
+                    </h3>
+                    <div class="youOrdersItem__value">
+                        48 hours
+                    </div>
+                </div>
+            </div>
+            <div class="headerButtons">
+                <button class="headerButtonsItem" type="submit">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: Check button color.

File: submissions/Zihfred/html5-form/index.html [hunk 4]
```diff
@@ -0,0 +1,591 @@
...
+                <div class="helpCenter__icon">
+                    <i class="far fa-question-circle"></i>
+                </div>
+                <div class="helpCenter__subtitle">
+                    If you need assistance while placing your order, contact one
+                    of our beloved Project Manager by phone or email.
+                </div>
+                <div class="helpCenter__contacts">
+                    <strong>
+                        <a href="tel:+971 (0) 555"
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: Add styles on `:hover`, `:active`, and `:focus` state.

File: submissions/Zihfred/html5-form/index.html [hunk 5]
```diff
@@ -0,0 +1,591 @@
...
+
+
+                </div>
+            </div>
+            <div class="paymentBlock numberedBlock">
+                <h1 class="selectLanguageBlock__title">
+                    Sign up and choose a payment method
+                </h1>
+                <h2 class="selectLanguageBlock__subtitle">
+                    Already have an account with us? Sign in.
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: `Sign in` must be a link.

File: submissions/Zihfred/html5-form/index.html [hunk 6]
```diff
@@ -0,0 +1,591 @@
...
+                </div>
+                <div class="cardInfo">
+
+                    <label
+                            for="cardInfo__item-number"
+                            class="cardInfo__item">
+                        <div class="registerForm__label--title">
+                            CARD NUMBER
+                        </div>
+                        <input
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: Check `background-position` of image. [Image](https://imgur.com/a/pTT8XwC)

File: submissions/Zihfred/html5-form/index.html [hunk 7]
```diff
@@ -0,0 +1,591 @@
...
+                                id="cardInfo__item-code"
+                                class="cardInfo__item--input cardInfo__item--inputCode"
+                                pattern="^[0-9]+$"
+                                minlength="3"
+                                maxlength="5"
+                                required>
+                    </label>
+
+                </div>
+                <div class="paymentBlock__buttons">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: This checkbox must be unchecked by default.

File: submissions/Zihfred/html5-form/index.html [hunk 8]
```diff
@@ -0,0 +1,591 @@
...
+                           id="paymentBlock__buttons--checkbox"
+                           checked
+                           required>
+                    <label
+                            for="paymentBlock__buttons--checkbox"
+                            class="paymentBlock__buttons--label">
+                        I agree to the Terms & Conditions
+                    </label>
+                    <div class="headerButtons">
+                        <button class="headerButtonsItem" type="submit">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: Check a color of the button.

File: submissions/Zihfred/html5-form/index.html [hunk 9]
```diff
@@ -0,0 +1,591 @@
...
+                            PLACE ORDER
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: Don't use text in uppercase as content. Use CSS for this. When you change text your visual representation should be the same and independent of text content or text case inside HTML.

File: submissions/Zihfred/html5-form/index.html [hunk 10]
```diff
@@ -0,0 +1,591 @@
...
+                    </strong>
+                </div>
+            </div>
+        </div>
+        <div class="insideWrapper">
+            <div class="selectTypeBlock">
+                <h1 class="selectTypeBlock__title">
+                    select what you want to translate
+                </h1>
+                <div class="selectTypeBlock__checkboxes">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: This gruop of radio-button is required.

File: submissions/Zihfred/html5-form/index.html [hunk 11]
```diff
@@ -0,0 +1,591 @@
...
+            <div class="selectLanguageBlock numberedBlock">
+                <h1 class="selectLanguageBlock__title">
+                    Choose your languages pairing
+                </h1>
+                <label for="selectLangFrom"
+                       class="langLabel">
+                    FROM
+                </label>
+                <div class="select">
+                    <select name="From"
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: This select is required.

File: submissions/Zihfred/html5-form/index.html [hunk 12]
```diff
@@ -0,0 +1,591 @@
...
+                        <option>Russian</option>
+                        <option>France</option>
+                    </select>
+                </div>
+                <br>
+                <label for="selectLangTo" class="langLabel ">
+                    TO
+                </label>
+                <div class="select">
+                    <select name="TO"
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: This select is required too.

File: submissions/Zihfred/html5-form/index.html [hunk 13]
```diff
@@ -0,0 +1,591 @@
...
+                            id="selectLangToSecond"
+                            class="selectLangFrom selectLang selectLangToSecond">
+                        <option>English</option>
+                        <option>Russian</option>
+                        <option>France</option>
+                    </select>
+
+                </div>
+                <div class="selectLangToSecond__icon">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: This must be `button`. Also, move it closer to the corresponding select.

File: submissions/Zihfred/html5-form/index.html [hunk 14]
```diff
@@ -0,0 +1,591 @@
...
+                <button class="selectLang__add" type="button">
+                    + ADD
+                </button>
+            </div>
+            <div class="wrapperInner">
+                <div class="selectDesired numberedBlock">
+                    <h1 class="selectDocumentBlock__title selectDesired__title">
+                        Select your desired tone
+                    </h1>
+                    <div class="selectDesired__checkboxes">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: Radio-buttons inside this block is required too.

File: submissions/Zihfred/html5-form/index.html [hunk 15]
```diff
@@ -0,0 +1,591 @@
...
+                            </div>
+                        </label>
+                    </div>
+                </div>
+                <div class="tellUrgent numberedBlock">
+                    <h1 class="selectDocumentBlock__title">
+                        Tell us how urgent this job is
+                    </h1>
+                    <div class="tellUrgent__range">
+                        <input type="range"
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: This input is required.

File: submissions/Zihfred/html5-form/index.html [hunk 16]
```diff
@@ -0,0 +1,591 @@
...
+                    </div>
+                    <div class="tellUrgent__timeline">
+                        <div class="timeline__info">
+                            <div class="timeline__info--icon">
+                                <i class="fas fa-info-circle"></i>
+                            </div>
+                            TIMELINE INFO
+                        </div>
+                        <div class="timeline__text">
+                            With an <span class="orange">AVERAGE</span> timeline set up our expert translator can take a
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: On the mockup this text in 2 lines.

File: submissions/Zihfred/html5-form/index.html [hunk 17]
```diff
@@ -0,0 +1,591 @@
...
+                        <i class="fas fa-trash-alt"></i>
+                    </a>
+                </div>
+                <button class="selectLang__add" type="button">
+                    + ADD
+                </button>
+            </div>
+            <div class="wrapperInner">
+                <div class="selectDesired numberedBlock">
+                    <h1 class="selectDocumentBlock__title selectDesired__title">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: Heading `h1`  must be only ONE on the page.

File: submissions/Zihfred/html5-form/index.html [hunk 18]
```diff
@@ -0,0 +1,591 @@
...
+                                required>
+                        <div class="registerForm__label--valid">
+                            <i class="fas fa-check"></i>
+                        </div>
+                        <div class="registerForm__label--invalid">
+                            <i class="fas fa-times"></i>
+                        </div>
+                    </label>
+                </div>
+                <div class="selectPayType">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: This block is required.

File: submissions/Zihfred/html5-form/index.html [hunk 19]
```diff
@@ -0,0 +1,591 @@
...
+                    Sign up and choose a payment method
+                </h1>
+                <h2 class="selectLanguageBlock__subtitle">
+                    Already have an account with us? Sign in.
+                </h2>
+                <div class="registerForm">
+                    <label
+                            for="registerForm__fullNameInput"
+                            class="registerForm__label registerForm__labelBlock">
+                        <div class="registerForm__label--title">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: Use `label` for this.

File: submissions/Zihfred/html5-form/index.html [hunk 20]
```diff
@@ -0,0 +1,591 @@
...
+                <div class="paymentBlock__buttons">
+                    <input type="checkbox"
+                           name="Agree"
+                           id="paymentBlock__buttons--checkbox"
+                           checked
+                           required>
+                    <label
+                            for="paymentBlock__buttons--checkbox"
+                            class="paymentBlock__buttons--label">
+                        I agree to the Terms & Conditions
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: `Term and conditions` must be a link.

File: submissions/Zihfred/html5-form/index.html [hunk 21]
```diff
@@ -0,0 +1,591 @@
...
+                    </label>
+
+                    <label
+                            for="cardInfo__item-date"
+                            class="cardInfo__item">
+                        <div class="registerForm__label--title">
+                            EXPIRE DATE
+                        </div>
+                        <input
+                                type="date"
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: This input is narrow. Make him wider, the year does not fit. [Image](https://imgur.com/a/GGqkR4B)

File: submissions/Zihfred/html5-form/index.html (hunk 3 above)
Comment (Zihfred) [reply submissions/Zihfred/html5-form/index.html]: Please, write more info or make screenshot

Comment (A-Ostrovnyy) [reply submissions/Zihfred/html5-form/index.html]: This is a call-to-action button on mockup it has been painted in an accent color.
On the mockup, it painted in primary color, in your version - without.

File: submissions/Zihfred/html5-form/index.html (hunk 19 above)
Comment (Zihfred) [reply submissions/Zihfred/html5-form/index.html]: now it's label `<label`

Comment (A-Ostrovnyy) [reply submissions/Zihfred/html5-form/index.html]: Ok.
If it's `.registerForm__label--title` class with BEM modificator there must be a class without BEM modificator.

File: submissions/Zihfred/html5-form/style/main.css [hunk 22]
```diff
@@ -688,61 +822,73 @@ input[type=range]:focus::-ms-fill-upper {
...
+.paymentBlock__buttons .headerButtons::after {
     display: none;
 }
-.headerButtons .headerButtonsItem:not(:last-child), .paymentBlock__buttons--label{
+
+.headerButtons .headerButtonsItem:not(:last-child),
+.paymentBlock__buttons--label {
     margin-right: 20px;
-}
+}
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/style/main.css]: Add an empty line at the end of every file.

File: submissions/Zihfred/html5-form/index.html [hunk 23]
```diff
@@ -334,9 +338,9 @@ <h1 class="selectDocumentBlock__title selectDesired__title">
                     </div>
                 </div>
                 <div class="tellUrgent numberedBlock">
-                    <h1 class="selectDocumentBlock__title">
+                    <div class="selectDocumentBlock__title">
                         Tell us how urgent this job is
-                    </h1>
+                    </div>
                     <div class="tellUrgent__range">
                         <input type="range"
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: Fix vertical alignment of slider thumb. [Image](https://imgur.com/a/Vv3idxl)

File: submissions/Zihfred/html5-form/index.html [hunk 24]
```diff
@@ -57,12 +57,12 @@ <h3 class="youOrdersItem__text">
                 </div>
             </div>
             <div class="headerButtons">
-                <button class="headerButtonsItem" type="submit">
-                    PLACE ORDER
+                <button class="headerButtonsItem headerButtonsItem--submit" type="submit">
```
Comment (A-Ostrovnyy) [inline submissions/Zihfred/html5-form/index.html]: This is call-to-action button. It must be painted in a primary/accent color. Same story with button at the end of the page.

**Comment by A-Ostrovnyy** [comment]
Great work!
Tablet view needs to be fixed:
- [Image](https://imgur.com/a/oHSuuoY)
- [Image](https://imgur.com/a/YLOVD63)
- [Image](https://imgur.com/a/BiOMEJD)

Also, add `margin-right` to the checkbox, and `text-decoration: underline;` to the links.

**Comment by A-Ostrovnyy** [comment]
Well done!

**Pull request merged**